btk/draw_blends.py

```diff
@@ -187,8 +184,13 @@ def __next__(self):
         for s in self.surveys:
             pix_stamp_size = int(self.stamp_size / s.pixel_scale)
 
-            # create WCS and PSF information
-            psf = [get_psf(s, filt, self.atmospheric_model) for filt in s.filters]
+            # make PSF and WCS
+            psf = []
+            for filt in s.filters:
+                if callable(filt.psf):
+                    psf.append(filt.psf(s, filt))
+                else:
+                    psf.append(filt.psf)
```

Comment:
maybe instead of `else` check if `felt.psf` is a galsim object and otherwise raise an error (or you could check it later since if `filt.psf` is callable you want to check it returns a galsim object too)

Reply:
```python
for filt in s.filters:
    ...
    psf.append(filt.psf(s, filt))
```
that piece of code looks really weird to me, in the sense that reading the code does not help understanding what's going on. It's a hint that some of it should be refactored.

Reply:
Yes.. In the callable case, maybe writing it in two lines like:
`psf_object = filt.psf(s, filt)`
`psf.append(psf_object)`
makes it a bit better. But it is also weird that the function that generates the PSF model needs the `Filter` as input parameter while it is itself a `Filter` parameter..

Reply:
I guess it will be refactored anyway with the changes discussed with @thuiop below

Reply:
It should become filt.psf().